src/utils/execution.ts

```diff
@@ -409,27 +409,28 @@ export const buildSafeTransaction = (template: {
   };
 };
 
-export async function buildecdsaModuleAuthorizedForwardTx(
+export async function buildEcdsaModuleAuthorizedForwardTx(
   destinationContract: string,
   callData: string,
   smartAccount: Contract,
   smartAccountOwner: Signer & TypedDataSigner,
-  moduleAddress: string,
-  value: number | string = 0,
+  validationModuleAddress: string,
+  forwardFlowModule: Contract,
```

Comment:
I'm trying to understand how two modules work together?
like you init with ECDSA module 
and how does this another enabled module - Forward module work? do we have some basic flow diagram?
when will the module manager methods execTransactionFromModule() will be triggered?

Reply:
So, the transaction is initiated at `ForwardFlowModule` by calling `execTransaction`.
Like it was previously `SmartAccount.execTransaction_S6w`.

Then ForwardFlowModule [calls SA.isValidSignature](https://github.com/bcnmy/scw-contracts/blob/7e4df55e10b82eb9fb9b9aba9774fde76b9d5706/contracts/smart-contract-wallet/modules/ForwardFlowModule.sol#L95-L100) to validate the signature.
SA forwards this to the corresponding Validation Module, which is ECDSA module.
And returns validation success to ForwardFlowModule

Then ForwardFlowModule [calls SmartAccount.execTransactionFromModule](https://github.com/bcnmy/scw-contracts/blob/7e4df55e10b82eb9fb9b9aba9774fde76b9d5706/contracts/smart-contract-wallet/modules/ForwardFlowModule.sol#L120-L126) to actually execute the transaction.

[handlePayment](https://github.com/bcnmy/scw-contracts/blob/7e4df55e10b82eb9fb9b9aba9774fde76b9d5706/contracts/smart-contract-wallet/modules/ForwardFlowModule.sol#L163) has also been rebuilt to work through `execTransactionFromModule`

Reply:
hmm so this is going to ECDSA auth module to verify it. 
https://github.com/bcnmy/scw-contracts/blob/7e4df55e10b82eb9fb9b9aba9774fde76b9d5706/contracts/smart-contract-wallet/modules/ForwardFlowModule.sol#L96

so essentially there is only one validation plugin / module. and here we required this auth module to call enableModule to enable Forward module.
I got confused because modules you enable (old gnosis way) can just call smart-account but they can have their own sig verification (which is forward validate signature here -> passed to main auth module :D ) [[ Also this sig here gets appended with forward module address as I saw , i think ]]

at the time there will only be one plugin? if yes/ even no how do you enable or replace auth modules?  [[ here you won't call enableModule right? like in init we said ok use this as auth module ]]

when we have social recovery module or session key module, they would be enabled via enableModule flow right? 
hence the other day I was saying modules are classified as auth/validation modules and execution modules (like we enabled forward) tell me If what i said makes sense?

Reply:
1. There can be any amount of modules enabled via enableModule at the same time.
2. Any amount of enabled modules can be Authorization Modules.
3. In AA flow, the userOp.signature is appended with Authorization module address, that is intended to validate this signature. Thus there can be ECDSA Auth Module, Passkey Auth Module, Session key Auth module enabled at the same time for a SA and userOps can be signed with any of those modules. If they are signed with an EOA, the userOp.signature should be appended with an ECDSA Auth module address, so the validation flow will be forwarded to the ECDSA Auth Module. If userOp is signed with passkey, the Passkey Auth module address should be appended etc. Validation module address is extracted from sig [here](https://github.com/bcnmy/scw-contracts/blob/7e4df55e10b82eb9fb9b9aba9774fde76b9d5706/contracts/smart-contract-wallet/SmartAccount.sol#L264-L266).
4. So the Auth Modules do not need to call `execTransactionFromModule`. They just validate the userOp sig, and execution is handled by EntryPoint calling SA.executeCall or any other method. This is an efficient AA flow.
5. Old-fashioned Gnosis style modules (let's call them exec modules) are also able to interact with SA through `execTransactionFromModule`. They can have their own Txn validation logic, but they shouldn't be called in an userOp (coz if you're doing thru userOp you should just use AA flow with Auth Module). They should be called directly. So they can do their own validation logic (or forward it to SA like ForwardFlow module) and then call `execTransactionFromModule`. However, this flow is inefficient and is only needed for one use case, when user explicitly wants Forward Flow -> ForwardFlowModule.